### PR TITLE
unrelated changes in merge conflict commit

### LINKED ISSUE #6349: Working directory changes committed unintentionally during merge conflict resolution
**Please describe the problem you think should be solved**

If I have changed files in my working directory prior to merging a different branch into mine (for example, updating my branch by merging in the latest from master) and I encounter merge conflicts, resolve the conflicts, and commit the merge, the files that were in my working directory will also be committed as part of the merge commit even though I likely did not intend for them to be.

This is likely confusing ("where did the stuff I was working on go?") and could be a disincentive to frequently staying up to date with the default branch, which we think is a good practice.

**Do you have any potential solutions in mind?**

My first inclination is that Desktop could ignore the working directory files during the merge and put them back after it's completed. I'm not sure what potential pitfalls exist with this.

Note: This was originally brought to my attention by @niik.

cc: @outofambit, @ampinsk, @nerdneha

### PR DESCRIPTION
**Pull request by outofambit**
## Overview

closes #6349 

## Description

- `createMergeCommit` and doesn't unstage anything prior to committing
- `_finishConflictedMerge` only manually stages files that are conflicted

Taken together, this means that we only commit files that git deems part of a conflicted merge (git automatically stages unconflicted files during a conflicted merge) as well as files that the user resolves (or stages manually). This means the user can wreak a little havoc on their merge commit for a conflicted merge, but only via the terminal, and only in very direct ways.

_NB_ — this diff is even smaller if you hide whitespace changes!

## Release notes

Notes: [Improved] Merge conflict resolution committing only commits changes relevant to the merge

### CODE REVIEW

File: app/test/helpers/repositories.ts [hunk 1]
```diff
@@ -120,6 +121,52 @@ export async function setupConflictedRepo(): Promise<Repository> {
...
+ * Setup a repository and create a merge conflict
+ *
+ * @returns the new local repository
+ *
+ * The current branch will be 'other-branch' and the merged branch will be
+ * 'master' in your test harness.
+ *
+ * The conflicted file will be 'foo'. There will also be uncommitted changes unrelated to the merge in 'perlin'.
+ */
+export async function setupConflictedRepoWithDirt(): Promise<Repository> {
```
Comment (say25) [inline app/test/helpers/repositories.ts]: More descriptive name
```suggestion
export async function setupConflictedRepoWithUnrelatedUncommittedChange(): Promise<Repository> {
```

**Comment by shiftkey** [comment]
👍 this is the same approach that I ended up going with for rebase flows

File: app/test/helpers/repositories.ts (hunk 1 above)
Comment (outofambit) [reply app/test/helpers/repositories.ts]: changed in 6b8647a

**Comment by outofambit** [comment]
Thanks @niik for your review! I totally agree we need to be careful about changes to how we ignore (or don't) the index. I went ahead and added the comment you mentioned as i agree that context will be important for us to preserve fo the future!

**Comment by outofambit** [comment]
merging this now in preparation for today's beta

**Pull request merged**